app/controllers/teams_controller.rb

```diff
@@ -106,4 +101,23 @@ def set_display_roles
       @display_roles = ['student']
       @display_roles.map!(&:pluralize)
     end
+
+    def show_teams_by(params)
```

Comment:
`params` is available as a controller method, you don't need to pass it as an argument. With the argument gone, I would go for a more generic method name, something like `find_teams`

It is also a veeeery long method.

I'm wondering why we only include the activities (w/ sorting) when there is a `:sort` param. I would think this way of sorting / eager loading makes sense whether or not we sort stuff. Do you know, @klappradla or @alicetragedy ?

Provided we allow combining all these options, we can just build up the query instead of having a complex `case` or `if … else` statement:

```ruby
direction = params[:direction] == 'asc' ? 'ASC' : 'DESC'
# start with the "base scope"
@teams = Team.by_season_phase.includes(:activities).order("teams.kind, activities.created_at #{direction}").references(:activities)
# Re-assign @teams if conditions apply:
@teams = @teams.by_season if foo
@teams = @teams.by_something_else if bar
```

Does that make sense?